aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -58,23 +69,221 @@
+
+    private final RemoveNotificationChannelRequest removeNotificationChannelRequest =
+            RemoveNotificationChannelRequest.builder()
+                    .profilingGroupName(profilingGroupName)
+                    .channelId("channelId")
+                    .build();
+
+    private final AddNotificationChannelsRequest addNotificationChannelsRequest = AddNotificationChannelsRequest.builder()
+            .profilingGroupName(profilingGroupName)
+            .channels(Channel.builder()
+                    .uri("channelUri2")
+                    .eventPublishers(ImmutableSet.of(EventPublisher.ANOMALY_DETECTION))
+                    .build())
+            .build();
 
     private final List<String> principals = Arrays.asList("arn:aws:iam::123456789012:role/UnitTestRole");
 
+    @Nested
+    class WhenNotificationChannelConfigurationIsProvided {
+        @BeforeEach
+        public void setup() {
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).anomalyDetectionNotificationConfiguration(
+                    Collections.singletonList(software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                            .channelUri("channelUri")
+                            .build()))
+                    .build());
+
+            doReturn(GetPolicyResponse.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+
+
+            doReturn(RemoveNotificationChannelResponse.builder().build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(removeNotificationChannelRequest), any());
+        }
+
+        @Test
+        public void testSuccess() {
+            doReturn(GetNotificationConfigurationResponse.builder()
+                    .notificationConfiguration(NotificationConfiguration.builder()
+                            .channels(Channel.builder()
+                                    .id("channelId")
+                                    .uri("channelUri")
+                                    .eventPublishers(EventPublisher.ANOMALY_DETECTION)
+                                    .build())
+                            .build())
+                    .build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+            final ProgressEvent<ResourceModel, CallbackContext> response
+                    = subject.handleRequest(proxy, request, null, logger);
+
+            assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+            assertThat(response.getCallbackContext()).isNull();
+            assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+            assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+            assertThat(response.getMessage()).isNull();
+            assertThat(response.getErrorCode()).isNull();
+        }
```

Comment:
If I'm reading it correctly, this test case validates the no-op scenario, right? That nothing gets changed?

Consider updating the naming of the test to reflect that, to make it easier to follow.

Reply:
:point_up_2: This one was resolved but has not been addressed.